Port NetworkResourceLoadParameters to the new IPC serialization format

gavin-apple opened the pull request:
#### db8f4c828416ab5cbe20155367ce24b8d2f987cf
<pre>
Port NetworkResourceLoadParameters to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=250201">https://bugs.webkit.org/show_bug.cgi?id=250201</a>
rdar://103951633

Reviewed by Alex Christensen.

Port NetworkResourceLoadParameters and other related types over to the
new IPC serialization format. This includes:
    * NetworkResourceLoadParameters
    * PreconnectOnly
    * SameSiteInfo
    * SecurityOrigin
    * ContentSniffingPolicy
    * ContentEncodingSniffingPolicy
    * ClientCredentialPolicy
    * ShouldRelaxThirdPartyCookieBlocking
    * PreflightPolicy
    * CrossOriginEmbedderPolicy
    * CrossOriginEmbedderPolicyValue
    * IncludeSecureCookies
    * CookieRequestHeaderFieldProxy

* Source/WebCore/Modules/fetch/FetchBody.cpp:
(WebCore::FetchBody::fromFormData):
* Source/WebCore/Modules/fetch/FetchBody.h:
* Source/WebCore/loader/CrossOriginEmbedderPolicy.cpp:
(WebCore::CrossOriginEmbedderPolicy::encode const):
(WebCore::CrossOriginEmbedderPolicy::decode):
* Source/WebCore/loader/CrossOriginEmbedderPolicy.h:
(WebCore::CrossOriginEmbedderPolicy::encode const): Deleted.
(WebCore::CrossOriginEmbedderPolicy::decode): Deleted.
* Source/WebCore/loader/NavigationRequester.h:
(WebCore::NavigationRequester::decode):
* Source/WebCore/loader/cache/KeepaliveRequestTracker.cpp:
(WebCore::KeepaliveRequestTracker::tryRegisterRequest):
(WebCore::KeepaliveRequestTracker::registerRequest):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::logMediaDiagnosticMessage const):
* Source/WebCore/page/Page.h:
* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::SecurityOrigin::create):
* Source/WebCore/page/SecurityOrigin.h:
(WebCore::SecurityOrigin::encode const): Deleted.
(WebCore::SecurityOrigin::decode): Deleted.
* Source/WebCore/platform/network/CookieRequestHeaderFieldProxy.h:
(WebCore::CookieRequestHeaderFieldProxy::encode const): Deleted.
(WebCore::CookieRequestHeaderFieldProxy::decode): Deleted.
* Source/WebCore/platform/network/FormData.cpp:
(WebCore::FormData::create):
* Source/WebCore/platform/network/FormData.h:
(WebCore::FormData::encode const): Deleted.
(WebCore::FormData::decode): Deleted.
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
(WebCore::ResourceRequestBase::httpBody const):
(WebCore::ResourceRequestBase::hasUpload const):
* Source/WebCore/platform/network/ResourceRequestBase.h:
* Source/WebCore/platform/network/SameSiteInfo.h:
(WebCore::SameSiteInfo::encode const): Deleted.
(WebCore::SameSiteInfo::decode): Deleted.
* Source/WebCore/platform/network/cf/FormDataStreamCFNet.cpp:
(WebCore::setHTTPBody):
* Source/WebCore/platform/network/cf/FormDataStreamCFNet.h:
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegate.cpp:
(WebCore::ResourceHandleCFURLConnectionDelegate::createResourceRequest):
* Source/WebCore/platform/network/cf/ResourceRequestCFNet.cpp:
(WebCore::ResourceRequest::doUpdatePlatformHTTPBody):
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::doUpdatePlatformHTTPBody):
* Source/WebCore/platform/network/curl/CurlFormDataStream.cpp:
(WebCore::CurlFormDataStream::CurlFormDataStream):
* Source/WebCore/platform/network/curl/CurlFormDataStream.h:
* Source/WebCore/platform/network/mac/FormDataStreamMac.h:
* Source/WebCore/platform/network/mac/FormDataStreamMac.mm:
(WebCore::setHTTPBody):
(WebCore::createHTTPBodyNSInputStream):
* Source/WebCore/platform/network/mac/ResourceHandleMac.mm:
(WebCore::ResourceHandle::willSendRequest):
* Source/WebCore/platform/network/soup/ResourceRequestSoup.cpp:
(WebCore::ResourceRequest::updateSoupMessageBody const):
* Source/WebCore/testing/ServiceWorkerInternals.cpp:
(WebCore::ServiceWorkerInternals::createOpaqueWithBlobBodyResponse):
* Source/WebCore/workers/service/context/ServiceWorkerFetch.cpp:
(WebCore::ServiceWorkerFetch::dispatchFetchEvent):
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/FormData.serialization.in: Added.
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::resolveBlobReferences):
* Source/WebKit/NetworkProcess/NetworkLoadParameters.h:
(WebKit::NetworkLoadParameters::NetworkLoadParameters):
* Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.cpp:
(WebKit::NetworkResourceLoadParameters::NetworkResourceLoadParameters):
(WebKit::NetworkResourceLoadParameters::parentOrigin const):
(WebKit::NetworkResourceLoadParameters::sandboxExtensionsIfHttpBody const):
(WebKit::NetworkResourceLoadParameters::sandboxExtensionIflocalFile const):
(WebKit::NetworkResourceLoadParameters::encode const): Deleted.
(WebKit::NetworkResourceLoadParameters::decode): Deleted.
* Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.h:
* Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.serialization.in: Added.
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::startNetworkLoad):
* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm:
(WebKit::NetworkDataTaskCocoa::willPerformHTTPRedirection):
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(-[WKNetworkSessionDelegate URLSession:task:needNewBodyStream:]):
* Source/WebKit/Platform/IPC/FormDataReference.h:
(IPC::FormDataReference::decode):
* Source/WebKit/Shared/URLSchemeTaskParameters.cpp:
(WebKit::URLSchemeTaskParameters::encode const):
(WebKit::URLSchemeTaskParameters::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;DOMCacheEngine::Record&gt;::encode):
(IPC::ArgumentCoder&lt;DOMCacheEngine::Record&gt;::decode):
(IPC::ArgumentCoder&lt;RefPtr&lt;SecurityOrigin&gt;&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;RefPtr&lt;SecurityOrigin&gt;&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;Ref&lt;SecurityOrigin&gt;&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;Ref&lt;SecurityOrigin&gt;&gt;::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/259142@main">https://commits.webkit.org/259142@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8390271634f06f705a262e7d7c010e9c7338c9ac

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104047 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13143 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36977 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113260 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173567 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/107992 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14201 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4055 "Built successfully") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96283 "Reverted pull request changes (failure)") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112342 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109818 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10946 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94012 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/36/builds/96283 "Reverted pull request changes (failure)") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92793 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25624 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/36/builds/96283 "Reverted pull request changes (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6512 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27001 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6657 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3536 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12654 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46535 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8434 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3338 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/platform/network/FormData.cpp:
```diff
@@ -91,6 +91,16 @@ Ref<FormData> FormData::create(const DOMFormData& formData, EncodingType encodin
     return result;
 }
 
+Ref<FormData> FormData::create(bool alwaysStream, Vector<char> boundary, Vector<WebCore::FormDataElement> elements, int64_t identifier)
```
These need to be Vector&&, otherwise it introduces unnecessary copies and allocations.

achristensen07 commented on Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.cpp:
```diff
@@ -31,406 +31,111 @@
-            if (auto handle = SandboxExtension::createHandleForReadByAuditToken(request.url().fileSystemPath(), *networkProcessAuditToken))
-                requestSandboxExtension = WTFMove(*handle);
-        } else
+NetworkResourceLoadParameters::NetworkResourceLoadParameters(WebCore::ResourceLoaderIdentifier identifier, WebPageProxyIdentifier webPageProxyID, WebCore::PageIdentifier webPageID, WebCore::FrameIdentifier webFrameID, WTF::ProcessID parentPID, WebCore::ResourceRequest resourceRequest, RefPtr<WebCore::FormData> httpBody, std::optional<Vector<SandboxExtension::Handle>> sandboxExtensionIfHttpBody, s ...
```
Can we remove all written-out constructors instead?  If not, can we put each parameter on its own line?  We usually don't, but for this many parameters it'll help with making git history more visible.

achristensen07 replied:
Ditto, remove or put each on own line.

achristensen07 commented on Source/WebKit/NetworkProcess/SameSiteData.serialization.in:
```diff
@@ -0,0 +1,27 @@
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+struct WebCore::SameSiteInfo {
```
Not a big deal, but why isn't this in WebCoreArgumentCoders?

achristensen07 commented on Source/WebKit/Shared/SecurityOrigin.serialization.in:
```diff
@@ -0,0 +1,35 @@
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+[Return=Ref] class WebCore::SecurityOrigin {
```
Ditto, why not WebCoreArgumentCoders?

achristensen07 commented on Source/WebCore/page/SecurityOrigin.cpp:
```diff
@@ -596,6 +596,23 @@ Ref<SecurityOrigin> SecurityOrigin::create(const String& protocol, const String&
     return origin;
 }
 
+Ref<SecurityOrigin> SecurityOrigin::create(WebCore::SecurityOriginData&& data, const String& domain, const String& filePath, const Markable<OpaqueOriginIdentifier, OpaqueOriginIdentifier::MarkableTraits>& opaqueOriginIdentifier, bool universalAccess, bool domainWasSetInDOM, bool canLoadLocalResources, bool enforcesFilePathSeparation, bool needsStorageAccessFromFileURLsQuirk, std::optional<bool> i ...
```
I'm curious why the SecurityOriginData is an r-value but domain and filePath aren't.  It think they should all be.

achristensen07 commented on Source/WebCore/loader/cache/KeepaliveRequestTracker.cpp:
```diff
@@ -55,7 +55,7 @@ bool KeepaliveRequestTracker::tryRegisterRequest(CachedResource& resource)
 {
     ASSERT(resource.options().keepAlive);
-    auto* body = resource.resourceRequest().httpBody();
+    RefPtr<FormData> body = resource.resourceRequest().httpBody();
```
auto or RefPtr.  RefPtr<FormData> is overly explicit.  Many times in this PR.

achristensen07 commented on Source/WebKit/CMakeLists.txt:
```diff
@@ -522,7 +522,9 @@ set(WebKit_SERIALIZATION_IN_FILES
     GPUProcess/media/MediaDescriptionInfo.serialization.in
     GPUProcess/media/TextTrackPrivateRemoteConfiguration.serialization.in
 
+    NetworkProcess/FormData.serialization.in
```
This file should either be in WebCoreArgumentCoders.serialization.in or in Source/WebKit/Shared.

achristensen07 commented on Source/WebCore/platform/network/FormData.cpp:
```diff
@@ -91,6 +91,16 @@ Ref<FormData> FormData::create(const DOMFormData& formData, EncodingType encodin
+{
+    auto result = create();
+    result->setAlwaysStream(alwaysStream);
+    result->m_boundary = boundary;
```
Needs WTFMove to prevent unnecessary Vector copy.  Same with elements below.

achristensen07 commented on Source/WebCore/platform/network/FormData.cpp:
```diff
@@ -91,6 +91,16 @@ Ref<FormData> FormData::create(const DOMFormData& formData, EncodingType encodin
     return result;
 }
 
+Ref<FormData> FormData::create(bool alwaysStream, Vector<char>&& boundary, Vector<WebCore::FormDataElement> elements, int64_t identifier)
```
Should be Vector&&

achristensen07 commented on Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.cpp:
```diff
@@ -31,406 +31,143 @@
+    NetworkLoadParameters&& networkLoadParameters
+    , WebCore::ResourceLoaderIdentifier identifier
+    , RefPtr<WebCore::FormData> httpBody
+    , std::optional<Vector<SandboxExtension::Handle>> sandboxExtensionIfHttpBody
```
Should be an r-value to prevent unnecessary vector copies, same with frameAncestorOrigins.  Using r-values with other members also reduces ref churn, such as httpBody, sandboxExtensionIflocalFile (which should also have capitalization fixed), parentFrameURL, frameURL, documentURL, openerURL, and mainDocumentURL.  It wouldn't hurt to make all of them r-values.

gavin-apple committed:
Port NetworkResourceLoadParameters to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=250201
rdar://103951633

Reviewed by Alex Christensen.

Port NetworkResourceLoadParameters and other related types over to the
new IPC serialization format. This includes:
    * NetworkResourceLoadParameters
    * PreconnectOnly
    * SameSiteInfo
    * SecurityOrigin
    * ContentSniffingPolicy
    * ContentEncodingSniffingPolicy
    * ClientCredentialPolicy
    * ShouldRelaxThirdPartyCookieBlocking
    * PreflightPolicy
    * CrossOriginEmbedderPolicy
    * CrossOriginEmbedderPolicyValue
    * IncludeSecureCookies
    * CookieRequestHeaderFieldProxy

* Source/WebCore/Modules/fetch/FetchBody.cpp:
(WebCore::FetchBody::fromFormData):
* Source/WebCore/Modules/fetch/FetchBody.h:
* Source/WebCore/loader/CrossOriginEmbedderPolicy.cpp:
(WebCore::CrossOriginEmbedderPolicy::encode const):
(WebCore::CrossOriginEmbedderPolicy::decode):
* Source/WebCore/loader/CrossOriginEmbedderPolicy.h:
(WebCore::CrossOriginEmbedderPolicy::encode const): Deleted.
(WebCore::CrossOriginEmbedderPolicy::decode): Deleted.
* Source/WebCore/loader/NavigationRequester.h:
(WebCore::NavigationRequester::decode):
* Source/WebCore/loader/cache/KeepaliveRequestTracker.cpp:
(WebCore::KeepaliveRequestTracker::tryRegisterRequest):
(WebCore::KeepaliveRequestTracker::registerRequest):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::logMediaDiagnosticMessage const):
* Source/WebCore/page/Page.h:
* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::SecurityOrigin::create):
* Source/WebCore/page/SecurityOrigin.h:
(WebCore::SecurityOrigin::encode const): Deleted.
(WebCore::SecurityOrigin::decode): Deleted.
* Source/WebCore/platform/network/CookieRequestHeaderFieldProxy.h:
(WebCore::CookieRequestHeaderFieldProxy::encode const): Deleted.
(WebCore::CookieRequestHeaderFieldProxy::decode): Deleted.
* Source/WebCore/platform/network/FormData.cpp:
(WebCore::FormData::create):
* Source/WebCore/platform/network/FormData.h:
(WebCore::FormData::encode const): Deleted.
(WebCore::FormData::decode): Deleted.
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
(WebCore::ResourceRequestBase::httpBody const):
(WebCore::ResourceRequestBase::hasUpload const):
* Source/WebCore/platform/network/ResourceRequestBase.h:
* Source/WebCore/platform/network/SameSiteInfo.h:
(WebCore::SameSiteInfo::encode const): Deleted.
(WebCore::SameSiteInfo::decode): Deleted.
* Source/WebCore/platform/network/cf/FormDataStreamCFNet.cpp:
(WebCore::setHTTPBody):
* Source/WebCore/platform/network/cf/FormDataStreamCFNet.h:
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegate.cpp:
(WebCore::ResourceHandleCFURLConnectionDelegate::createResourceRequest):
* Source/WebCore/platform/network/cf/ResourceRequestCFNet.cpp:
(WebCore::ResourceRequest::doUpdatePlatformHTTPBody):
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::doUpdatePlatformHTTPBody):
* Source/WebCore/platform/network/curl/CurlFormDataStream.cpp:
(WebCore::CurlFormDataStream::CurlFormDataStream):
* Source/WebCore/platform/network/curl/CurlFormDataStream.h:
* Source/WebCore/platform/network/mac/FormDataStreamMac.h:
* Source/WebCore/platform/network/mac/FormDataStreamMac.mm:
(WebCore::setHTTPBody):
(WebCore::createHTTPBodyNSInputStream):
* Source/WebCore/platform/network/mac/ResourceHandleMac.mm:
(WebCore::ResourceHandle::willSendRequest):
* Source/WebCore/platform/network/soup/ResourceRequestSoup.cpp:
(WebCore::ResourceRequest::updateSoupMessageBody const):
* Source/WebCore/testing/ServiceWorkerInternals.cpp:
(WebCore::ServiceWorkerInternals::createOpaqueWithBlobBodyResponse):
* Source/WebCore/workers/service/context/ServiceWorkerFetch.cpp:
(WebCore::ServiceWorkerFetch::dispatchFetchEvent):
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/FormData.serialization.in: Added.
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::resolveBlobReferences):
* Source/WebKit/NetworkProcess/NetworkLoadParameters.h:
(WebKit::NetworkLoadParameters::NetworkLoadParameters):
* Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.cpp:
(WebKit::NetworkResourceLoadParameters::NetworkResourceLoadParameters):
(WebKit::NetworkResourceLoadParameters::parentOrigin const):
(WebKit::NetworkResourceLoadParameters::sandboxExtensionsIfHttpBody const):
(WebKit::NetworkResourceLoadParameters::sandboxExtensionIflocalFile const):
(WebKit::NetworkResourceLoadParameters::encode const): Deleted.
(WebKit::NetworkResourceLoadParameters::decode): Deleted.
* Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.h:
* Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.serialization.in: Added.
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::startNetworkLoad):
* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm:
(WebKit::NetworkDataTaskCocoa::willPerformHTTPRedirection):
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(-[WKNetworkSessionDelegate URLSession:task:needNewBodyStream:]):
* Source/WebKit/Platform/IPC/FormDataReference.h:
(IPC::FormDataReference::decode):
* Source/WebKit/Shared/URLSchemeTaskParameters.cpp:
(WebKit::URLSchemeTaskParameters::encode const):
(WebKit::URLSchemeTaskParameters::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<DOMCacheEngine::Record>::encode):
(IPC::ArgumentCoder<DOMCacheEngine::Record>::decode):
(IPC::ArgumentCoder<RefPtr<SecurityOrigin>>::encode): Deleted.
(IPC::ArgumentCoder<RefPtr<SecurityOrigin>>::decode): Deleted.
(IPC::ArgumentCoder<Ref<SecurityOrigin>>::encode): Deleted.
(IPC::ArgumentCoder<Ref<SecurityOrigin>>::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/259142@main
